src/__test__/components/data-exploration/heatmap/HeatmapPlot.test.jsx

```diff
@@ -1,545 +1,163 @@
+jest.mock('object-hash', () => {
+  const objectHash = jest.requireActual('object-hash');
+  const mockWorkResultETag = jest.requireActual('__test__/test-utils/mockWorkResultETag').default;
 
-  it('renders Vega heatmap when genes loaded', () => {
-    const store = mockStore(initialState);
-    component = mount(
-      <Provider store={store}>
-        <HeatmapPlot experimentId={experimentId} width={200} height={200} />
-      </Provider>,
-    );
-    expect(component.find('HeatmapPlot').length).toEqual(1);
-    expect(component.find('VegaHeatmap').length).toEqual(1);
-  });
+  const mockWorkRequestETag = (ETagParams) => `${ETagParams.body.nGenes}-marker-genes`;
+  const mockGeneExpressionETag = (ETagParams) => `${ETagParams.missingGenesBody.genes.join('-')}-expression`;
 
-  it('renders error state when expression data errors out', () => {
-    const store = mockStore({
-      ...initialState,
-      genes: {
-        ...initialState.genes,
-        expression: {
-          ...initialState.genes.expression,
-          error: 'wow, error!',
-        },
-      },
-    });
+  return mockWorkResultETag(objectHash, mockWorkRequestETag, mockGeneExpressionETag);
+});
 
-    component = mount(
-      <Provider store={store}>
-        <HeatmapPlot experimentId={experimentId} width={200} height={200} />
-      </Provider>,
-    );
+jest.mock('utils/work/seekWorkResponse', () => ({
+  __esModule: true,
+  seekFromAPI: jest.fn(),
+  seekFromS3: () => Promise.resolve(null),
+}));
 
-    expect(component.find('HeatmapPlot').length).toEqual(1);
-    expect(component.find(Empty).length).toEqual(1);
-  });
+jest.mock('next/dynamic', () => () => () => 'Sup Im a heatmap');
 
-  it('renders error state when the view errors out', () => {
-    const store = mockStore({
-      ...initialState,
-      genes: {
-        ...initialState.genes,
-        expression: {
-          ...initialState.genes.expression,
-          views: {
-            ...initialState.genes.expression.views,
-            [componentType]: {
-              ...initialState.genes.expression.views[componentType],
-              error: true,
-            },
-          },
-        },
-      },
-    });
+// // Worker responses are fetched from S3, so these endpoints are added to fetchMock
+// // the URL for the endpoints are generated by the functions passed to mockETag above
+const mockWorkerResponses = {
+  '5-marker-genes': () => Promise.resolve(_.cloneDeep(markerGenesData5)),
+  '2-marker-genes': () => Promise.resolve(_.cloneDeep(markerGenesData2)),
+};
 
-    component = mount(
-      <Provider store={store}>
-        <HeatmapPlot experimentId={experimentId} width={200} height={200} />
+let storeState = null;
+const loadAndRenderDefaultHeatmap = async () => {
+  storeState = makeStore();
+
+  storeState.dispatch(loadProcessingSettings(experimentId));
+  storeState.dispatch(loadBackendStatus(experimentId));
```

Comment:
why do you need these for the heatmap to work?

Reply:
processingSettings to know the resolution of configure embedding, which is sent in the worker task.
backend status is used in the `fetchWork` call

Reply:
indeed, that's for the marker heatmap to work. Thanks!